enhancements/insights/up-to-date-gathering.md

```diff
@@ -0,0 +1,250 @@
+---
+title: insights-up-to-date-gathering
+authors:
+  - "@martinkunc"
+  - "@inecas"
+reviewers:
+  - "@smarterclayton"
+  - "@mfojtik"
+  - "@jhjaggars"
+approvers:
+  - "@smarterclayton"
+  - "@mfojtik"
+  - "@jhjaggars"
+creation-date: 2020-08-21
+last-updated: 2020-08-21
+status: provisional
+---
+
+# Up to date gathering in Insights operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this change is to evolve Insights Operator to make the data gathering component less dependent 
+on the the particular release of the cluster. The proposal also suggest to follow more the standard operator 
+pattern, where the responsibility of the main operator process would shift from gathering the data itself to 
+watch over the gathering process, such as the selection of desired image stream or the fallback mechanism in 
+case something goes wrong.
+
+![](io-gatherer-with-container.png)
+
+## Motivation
+
+The motivation is to be able to react faster on requirements for additional
+Insights Operator data requests. The benefit for customers would be faster
+identification of clusters based on newly discovered issues, identified by
+particular cluster configuration.
+The other potential slowdown might be caused by fact that administrators need
+to upgrade their clusters to deliver actual updates (either automatically or manually).
+
+Example of situation when the process is too slow:
+Two weeks ago we discovered an issue that for larger amount of generated alerts, IO Archives
+becomes too large (#148, CCXDEV-2673). We decided to fix it in all available versions (4.6, - 4.3)
```

Comment:
What is 148?  I assume the CCXDEV reference is private, too.  Can you update this doc with publicly available references?